frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts

```diff
@@ -128,3 +121,20 @@ export const InventoryItemLoading: React.FC = require('@console/shared/src/compo
   .InventoryItemLoading;
 
 export { useFlag } from '../utils/flags';
+
+/**
+ * A lazy component that creates a monaco editor
+ * @example
+ * ```tsx
+ * <ResourceYAMLEditor
+ *     content={yamlString}
+ *     header="Create resource"
+ *     onSave={(content) => updateResource(content)}
+ *    />
+ * ```
+ * @param {ResourceYAMLEditorProps['content']} content - Stringified YAML/Object to be shown by the Editor
```

Comment:
```suggestion
 * @param {ResourceYAMLEditorProps['content']} content - YAML representing a resource to be shown by the editor
```

Reply:
The component accepts both object and YAML string. I guess most users would find it easy to just pass the object they receive from `k8sGet`

Reply:
Sorry, I didn't realize the component could take the object as well. `content` is probably the wrong name in that case :(

Also I assume the YAML editor only looks at this on load. If so, I'd prefix the prop with `initial` to make it clear the editor won't replace its contents.